src/foam/u2/crunch/wizardflow/CheckNoDataAgent.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.u2.crunch.wizardflow',
+  name: 'CheckNoDataAgent',
+
+  implements: [
+    'foam.core.ContextAgent',
+    'foam.mlang.Expressions'
+  ],
+
+  imports: [
+    'capabilities',
+    'crunchService',
+    'endSequence',
+    'rootCapability'
+  ],
+
+  exports: [
+    'submitted'
+  ],
+
+  properties: [
+    {
+      class: 'Boolean',
+      name: 'submitted'
+    }
+  ],
+
+  methods: [
+    // If Property expressions ever unwrap promises this method can be blank.
+    async function execute() {
+      var shouldOpen = false;
+
+      await this.crunchService.getGrantPath(null, this.rootCapability.id).then(capabilitiesNeeded => {
+        capabilitiesNeeded.forEach(capa => {
+          if ( capa.of ){
+            shouldOpen = true;
+          }
+
+          if ( ! shouldOpen ) {
```

Comment:
This looks like it should be after the forEach

Reply:
true